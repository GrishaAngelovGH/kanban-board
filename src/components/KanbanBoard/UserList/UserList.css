.user-list {
  height: 230px;
}