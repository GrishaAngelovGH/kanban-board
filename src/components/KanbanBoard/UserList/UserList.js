import { useState, useEffect } from "react"

import ListGroup from "react-bootstrap/ListGroup"

import users from "users"

const User = ({ id, name, image, selected, onSelect }) => {
  const [isSelected, setIsSelected] = useState(selected)

  const handleToggle = () => {
    setIsSelected(!isSelected)
    onSelect(!isSelected, id)
  }

  return (
    <div className="row align-items-center" onClick={handleToggle}>
      <div className="col-2">
        <img src={image} width={50} className="rounded-circle" alt="assigned-user" />
      </div>
      <div className="col-8">
        <p className="m-0 fw-bold">{name}</p>
      </div>
      <div className="col-2">
        {isSelected && (<i className="bi bi-check-circle-fill text-success fs-4"></i>)}
      </div>
    </div>
  )
}

const UserList = ({ task, onUpdate }) => {
  const [ids, setIds] = useState(task.assignedIds)

  const handleSelectUser = (shouldAdd, id) => {
    if (shouldAdd) {
      setIds([...ids, id])
    }

    if (!shouldAdd) {
      setIds(ids.filter(v => v !== id))
    }
  }

  useEffect(() => {
    onUpdate(ids)
  }, [ids, onUpdate])

  return (
    <ListGroup>
      {
        users.map(v => {
          const selected = ids.includes(v.id)

          return (
            <ListGroup.Item key={v.id} action>
              <User
                id={v.id}
                name={v.name}
                image={v.image}
                selected={selected}
                onSelect={handleSelectUser}
              />
            </ListGroup.Item>
          )
        })
      }
    </ListGroup>
  )
}

export default UserList