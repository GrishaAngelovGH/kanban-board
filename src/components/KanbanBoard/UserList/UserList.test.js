import { render } from "@testing-library/react"

import UserList from "./UserList"

test("should render UserList component", () => {
  const task = { id: 1, title: "Title", description: "Description", assignedIds: [] }
  const onUpdate = jest.fn()

  const view = render(<UserList task={task} onUpdate={onUpdate} />)

  expect(view).toMatchSnapshot()
})