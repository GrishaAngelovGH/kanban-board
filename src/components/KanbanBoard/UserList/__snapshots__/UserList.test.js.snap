// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="list-group"
      >
        <button
          class="list-group-item list-group-item-action"
        >
          <div
            class="row align-items-center"
          >
            <div
              class="col-2"
            >
              <img
                alt="assigned-user"
                class="rounded-circle"
                src="image_1"
                width="50"
              />
            </div>
            <div
              class="col-8"
            >
              <p
                class="m-0 fw-bold"
              >
                John Smith
              </p>
            </div>
            <div
              class="col-2"
            />
          </div>
        </button>
        <button
          class="list-group-item list-group-item-action"
        >
          <div
            class="row align-items-center"
          >
            <div
              class="col-2"
            >
              <img
                alt="assigned-user"
                class="rounded-circle"
                src="image_2"
                width="50"
              />
            </div>
            <div
              class="col-8"
            >
              <p
                class="m-0 fw-bold"
              >
                Jane Smith
              </p>
            </div>
            <div
              class="col-2"
            />
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="list-group"
    >
      <button
        class="list-group-item list-group-item-action"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-2"
          >
            <img
              alt="assigned-user"
              class="rounded-circle"
              src="image_1"
              width="50"
            />
          </div>
          <div
            class="col-8"
          >
            <p
              class="m-0 fw-bold"
            >
              John Smith
            </p>
          </div>
          <div
            class="col-2"
          />
        </div>
      </button>
      <button
        class="list-group-item list-group-item-action"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col-2"
          >
            <img
              alt="assigned-user"
              class="rounded-circle"
              src="image_2"
              width="50"
            />
          </div>
          <div
            class="col-8"
          >
            <p
              class="m-0 fw-bold"
            >
              Jane Smith
            </p>
          </div>
          <div
            class="col-2"
          />
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
